pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -222,6 +196,7 @@ func newRepositoryWithClient(
 		Repository: repo,
 
 		ctx:                    ctx,
+		app:                    app,
```

Comment:
Why making it part of this struct if it is already contained in the context. Accessing the whole app from inside of middleware shouldn't be that easy.